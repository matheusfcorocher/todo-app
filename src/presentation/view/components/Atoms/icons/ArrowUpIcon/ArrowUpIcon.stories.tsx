import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import ArrowUpIcon from './ArrowUpIcon';

export default {
  title: 'Atoms/ArrowUpIcon',
  component: ArrowUpIcon,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0 }}>
        <Story />
      </ul>
    ),
  ]
} as ComponentMeta<typeof ArrowUpIcon>;

const Template: ComponentStory<typeof ArrowUpIcon> = (args) => <ArrowUpIcon />;

export const Default = Template.bind({});
Default.args = {};

Default.args = {
};