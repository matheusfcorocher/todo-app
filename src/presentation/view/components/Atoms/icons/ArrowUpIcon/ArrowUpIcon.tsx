import React, { useState } from "react";

interface ArrowUpIconProps {
  className?: string;
}

function ArrowUpIcon({ className }: ArrowUpIconProps) {
  return (
    <svg
      role="image"
      className={`${className}`}
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      preserveAspectRatio="none"
    >
      <path fill="none" d="M0 0h24v24H0V0z" />
      <path fill="" d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z" />
    </svg>
  );
}

export default ArrowUpIcon;
