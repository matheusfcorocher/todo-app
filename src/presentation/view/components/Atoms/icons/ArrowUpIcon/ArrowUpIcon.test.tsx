import React, { useState } from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ArrowUpIcon from './ArrowUpIcon';

describe("::Components ::Atoms ::ArrowUpIcon", () => {
    test('renders icon of ArrowUpIcon', () => {
        render(<ArrowUpIcon />);
        const icon = screen.getByRole("image");
        expect(icon).toMatchSnapshot();
    })
})
