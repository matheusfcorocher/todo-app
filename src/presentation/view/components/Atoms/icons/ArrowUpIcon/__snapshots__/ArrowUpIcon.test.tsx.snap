// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`::Components ::Atoms ::ArrowUpIcon renders icon of ArrowUpIcon 1`] = `
<svg
  class="undefined"
  preserveAspectRatio="none"
  role="image"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h24v24H0V0z"
    fill="none"
  />
  <path
    d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
    fill=""
  />
</svg>
`;
